app/components/primer/clipboard_copy.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use ClipboardCopy to populate copy values from the DOM
+  class ClipboardCopy < Primer::Component
+    status :alpha
+
+    # Optional target element that holds the data the user will copy.
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :target, lambda { |**system_arguments|
+      system_arguments[:tag] ||= :div
+      system_arguments[:id] = @id
+
+      Primer::BaseComponent.new(**system_arguments)
+    }
+
+    # @example Default
+    #   <%= render(Primer::ClipboardCopy.new(value: "Text to copy")) do %>
+    #     "Click to copy!"
+    #   <% end %>
+    #
+    # @example With a target
+    #   <%= render(Primer::ClipboardCopy.new(id: "foo")) do |component|
```

Comment:
Do we need to have the consumer provide an `id` at all, or could we generate one randomly for them? (Or maybe both?

Reply:
I thought about randomly generating an ID. I didn't want to complicate too much, but it's certainly possible for us to do that. We want to support the user providing an ID to link any DOM node they want to the clipboard copy element.

I'll add some code to generate IDs automatically when not provided by the user.